plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useTinyEditor } from '../hooks/useTinyEditor';
+
+type WooApiMessage = {
+	role: string;
+	content: string;
+};
+
+type WooApiResponse = {
+	generated_text: string;
+	previous_messages: WooApiMessage[];
+};
+
+const generatingContentPhrases = [
+	__( 'Please wait…', 'woocommerce' ),
+	__( 'Just a little longer…', 'woocommerce' ),
+	__( 'Almost there…', 'woocommerce' ),
+];
+
+const getGeneratingContentPhrase = () =>
+	generatingContentPhrases[
+		Math.floor( Math.random() * generatingContentPhrases.length )
+	];
+
+export function DescriptionButtonContainer() {
+	const [ fetching, setFetching ] = useState( false );
+	const generatingContentPhraseInterval = useRef< number >();
+	const titleEl = useRef< HTMLInputElement >(
+		document.querySelector( '#title' )
+	);
+	const [ productTitle, setProductTitle ] = useState< string >(
+		titleEl.current?.value || ''
+	);
+	const tinyEditor = useTinyEditor();
+
+	useEffect( () => {
+		titleEl.current?.addEventListener( 'keyup', ( e ) =>
+			setProductTitle( ( e.target as HTMLInputElement ).value || '' )
+		);
+	}, [ titleEl ] );
+
+	useEffect( () => {
+		if ( fetching ) {
+			tinyEditor.setContent( getGeneratingContentPhrase() );
+			generatingContentPhraseInterval.current = setInterval(
+				() => tinyEditor.setContent( getGeneratingContentPhrase() ),
+				2000
+			);
+		} else {
+			clearInterval( generatingContentPhraseInterval.current );
+		}
+	}, [ fetching ] );
+
+	const buildPrompt = () => {
+		const instructions = [
```

Comment:
This looks like a good start. 👍

Reply:
Nitpick: for performance, we could memo-ize buildPrompt():

```
const buildPrompt = useMemo(() => {
    const instructions = [
        `Write a product description with the following product title: "${productTitle}".`,
        'Use a 9th grade reading level.',
        `Make the description ${DESCRIPTION_MAX_LENGTH} words or less.`,
        'Structure the description using standard HTML paragraph, strong and list tags.',
        'Do not include a heading at the very top of the description.',
    ];

    return instructions.join('\n');
}, [productTitle]);
```
When the component re-renders and productTitle is the same, it can use this memoized version of the function.
The performance impact right now is probably negligible but in the future we may perform some operations here (to get their personalized style, for instance).

Reply:
Good thought, although I agree with Lauren here -- I tend to shy away from adding more complexity for this kind of thing until there is something more taxing going on. Kent Dodds actually [has a blog about this](https://kentcdodds.com/blog/usememo-and-usecallback).